### PR TITLE
Visual studio code workspace

### PR DESCRIPTION
**Pull request by marcellourbani**
I guess we don't want to endorse any tool, but this will make it much easier for a pretty common one and won't hurt much others
With this a novice opening the folder will be

- offered to open it as a workspace
- if accepts, will be prompted to install the purescript extensions
- once these are in place, tooling will work out of the box once the projects are built (and get a prompt to build one if they're not)

### CODE REVIEW

File: purescriptbook.code-workspace [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    },
+    {
+      "path": "exercises/chapter14"
+    },
+    {
+      "path": "tmp/purcel"
+    },
+    {
+      "path": "tmp/purescript-comonad-ui-todos"
+    }
```
Comment (milesfrain) [inline purescriptbook.code-workspace:50]: Are these lines intended to be part of this PR?

**Comment by milesfrain** [comment]
The `.code-workspace` concept is new to me, and I'm a bit confused how this should work. Testing this PR locally and reading: https://code.visualstudio.com/docs/editor/workspaces#_multiroot-workspaces

I'm not noticing a difference when opening a project with or without the `.code-workspace` file in this PR.

Is this PR supposed to eliminate messages like these?:
> It doesn't look like the workspace root is a PureScript project. The PureScript project should be opened as a root workspace folder.

image

File: purescriptbook.code-workspace [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      "exercises": true
+    }
+  },
+  "extensions": {
+    "recommendations": [
+      "nwolverson.ide-purescript",
+      "nwolverson.language-purescript",
+      "mvakula.vscode-purty"
+    ]
+  }
```
Comment (milesfrain) [inline purescriptbook.code-workspace:68]: I only have the first two extensions installed. Should the purty extension be recommended if I open a project with PR applied?

File: purescriptbook.code-workspace (hunk 1 above)
Comment (marcellourbani) [reply purescriptbook.code-workspace:50]: Npm my mistake, will send a fix, sorry

File: purescriptbook.code-workspace (hunk 2 above)
Comment (marcellourbani) [reply purescriptbook.code-workspace:68]: When you open the folder you should get a prompt to open the workspace.
If you say yes or open directly you will get a prompt to install the extensions.
And opening a purescript file will give you a prompt to build the subproject. When built language support works. With some bugs supporting multiple folders at the same time but better than nothing

Comment (marcellourbani) [reply purescriptbook.code-workspace:62]: I cleaned up the workspace.
After opening a file with the workspace:
image
...and after building:
image

**Comment by milesfrain** [comment]
> When you open the **folder** you should get a prompt to open the workspace.

Which folder? I tried both opening the root folder and a specific chapter folder, and in both cases I'm noting seeing any _workspace_ prompts.
```
➜  psbe git:(vscode-workspace) code --version             
1.59.0
379476f0e13988d90fab105c5c19e7abc8b1dea8
x64
➜  psbe git:(vscode-workspace) ls
book.toml  CONTRIBUTING.md  deploy_key.enc  exercises  purescriptbook.code-workspace  README.md  scripts  text
➜  psbe git:(vscode-workspace) code .        
➜  psbe git:(vscode-workspace) code exercises/chapter2
```

Could you share a few more details on how to reproduce the undesired original behavior, and what I should expected to see with this `.code-workspace` addition? I'm having trouble observing a difference on my system.

**Comment by marcellourbani** [comment]
Sorry about this Miles
This is what I get after opening the folder:
```shell
code purescript-book/ 
```
image
I guess you don't get it because you disabled it in your configuration

and if I open the workspace file it gets in the workspace directly
Anyway you can always open it explicitly from the menu
```shell
code purescript-book/purescriptbook.code-workspace
```
image

Once you opened the workspace, language server works:
image

**Comment by milesfrain** [comment]
Thanks. I was only able to see how this PR works by re-cloning to a _separate_ directory.

Looks like there are some global vscode setting stored in `~/.config/Code` that track workspace settings on a per-directory basis, so a `git clean -xdff` (or even re-cloning to the same path) doesn't trigger the workspace prompt that I expected to see with this PR.

Thanks for providing all the additional examples and helping me figure this out.

**Pull request merged**